### PR TITLE
Stop bucketing users who aren't part of an experiment.

### PR DESCRIPTION
**Pull request by telaviv**
We were bucketing users who were not part of an experiment. Mostly due to a bad truthy check.

👓  @uzi @arjunb023 @prashtx

### CODE REVIEW

File: src/app/featureFlags.js [hunk 1]
```diff
@@ -345,7 +345,7 @@ const firstBuckets = new Set();
 flags.addRule('variant', function (name) {
   const [experiment_name, checkedVariant] = name.split(':');
   const experimentData = getExperimentData(this.state, experiment_name);
-  if (experimentData) {
+  if (Object.keys(experimentData).length !== 0) { // can be an empty object.
```
Comment (prashtx) [inline src/app/featureFlags.js]: Is it better to return `null` from `getExperimentData`? It sounds like it's always an error to access properties of the return value of `getExperimentData` when the current user is not part of the experiment in question. That would be more straightforward than checking the number of keys and would alert future consumers via an error if they've used this function counter to your intention.

**Comment by prashtx** [comment]
💅

**Comment by telaviv** [comment]
Two more feature requests here.

1) We fixed the bucketing issues surrounding require_login. We should only fire bucketing events if the user is both logged out and hasn't dismissed the event. Note I couldn't get this to work for the transparent experiment because we need to check the transparent bucket to know if we should even fire an xpromo view/ineligiblity event. 

Since we are purposefully trying not check variants, we also will be missing experiment data for the xpromo ineligiblity events. There's a weird catch 22 here. We can't know what experiment we're in unless we check it. We can't check it without fire a bucketing event. The bucketing event, at the moment, means they saw it. So we're a bit trapped with the current feature flag system.

2) Changed the experiment names. 

💇

**Comment by telaviv** [comment]
💇 removed our control experiment

File: src/app/components/DualPartInterstitial/Footer/index.jsx [hunk 2]
```diff
@@ -135,10 +141,8 @@ const selector = createSelector(
...
   requireXPromoLogin,
   state => features.withContext({ state }),
-  state => state.user.loggedOut,
   state => linkType => createNativeAppLink(state, linkType),
-  (subredditName, requireXPromoLogin, featureContext, loggedOut, createLink) => {
-    const requireLogin = loggedOut && requireXPromoLogin;
+  (subredditName, requireLogin, featureContext, createLink) => {
     const nativeInterstitialLink = createLink('interstitial');
     const nativeLoginLink = createLink('login');
     return { subredditName, requireLogin, nativeInterstitialLink, nativeLoginLink };
```
Comment (uzi) [inline src/app/components/DualPartInterstitial/Footer/index.jsx]: Do we still intend on returning `requireLogin` which is not undefined?  Can we explicitly set it if so?

Comment (uzi) [inline src/app/components/DualPartInterstitial/Footer/index.jsx]: Oh wait... brain fart...

Comment (uzi) [inline src/app/components/DualPartInterstitial/Footer/index.jsx]: Change that... do we need featureContext here?  Looks unused.

**Comment by telaviv** [comment]
💇

**Comment by telaviv** [comment]
i missed the experiment name
:haircut_man:

**Comment by uzi** [comment]
:fish:

**Comment by telaviv** [comment]
:haircut_man:

**Comment by uzi** [comment]
:fish: assuming tests pass

**Comment by telaviv** [comment]
:haircut_man:

**Comment by uzi** [comment]
:fish: more fishes for you

**Pull request merged**